azure-kusto-ingest/source/descriptors.js

```diff
@@ -65,16 +73,14 @@ class FileDescriptor {
 
 
 class StreamDescriptor {
-    constructor(stream, sourceId = null, isCompressed = false) {
+    constructor(stream, sourceId = null, compressionType) {
```

Comment:
this will still have a default value of `undefined`, because javascript allows to omit part of the params like:

```js
function foo(a,b,c) {
   console.log(a,b,c);
}

foo(1);
// will print 1,undefined ,undefined 
foo(1,2,3);
// will print 1,2,3
```

This is why ordering matters, cause to make it make sense, you leave the optionals for last, otherwise you are left with an annoying syntax like:
`func(1, null, 2);`

And since we don't want a breaking change, I would advise thinking of a default value + proper handling).